airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -283,7 +283,14 @@ static void watchForOrphanThreads(final Procedure runMethod,
    *         provided message.
    */
   @VisibleForTesting
-  static void consumeMessage(final AirbyteMessageConsumer consumer, final String inputString) throws Exception {
+  static void consumeMessage(final AirbyteMessageConsumer consumer, final String inputString, final long runnerMaximumMemoryInBytes)
+      throws Exception {
+
+    if (inputString.getBytes().length > (runnerMaximumMemoryInBytes * 0.6)) {
+      LOGGER.error("One messge is too big, the replication will fail in order to avoid OOM. The size of the message is: {} bytes",
+          inputString.getBytes().length);
+      throw new IllegalStateException("Invalid message, the message is too big");
```

Comment:
I think this error will eventually bubble up to the user/UI. If that's true, perhaps we should provide a little more context to try and learn as much as we can about the message:

```
Airbyte has received a message from the source(!) at 2022-01-01 01:02:03(!) which is larger than xxxMB(!).  The sync has been failed to prevent running out of memory.
```

Everything with (!) is dynamic.  Ideally, I'd like to use a regular expression or similar to grab the stream name, but we can add that later if this becomes a problem.

Reply:
Will this exception properly be attributed to the source and build a FailureReason?

Reply:
I'll check that.

Reply:
During my test, it turns out that adding that here will only be apply on the destination. I'll rework it and test the error message and make sure that the getMaxMemory is only call once.

Reply:
@evantahler , I made some change (it's a draft because I need to clean it up, I'll re-request a review when its ready). In my test it looked like 
<img width="1517" alt="image" src="https://user-images.githubusercontent.com/6792130/211442371-df0189f1-2ee5-46e6-badb-c7c58e2368bd.png">

Reply:
Nice!